src/components/common/data/DataView.tsx

```diff
@@ -0,0 +1,180 @@
+  columns: Array<TableColumnType>;
+  filterStatement: string;
+  handleChangeTable: () => void;
+  handleDrawerWidth: () => void;
+  pageBQQuery: () => void;
+  panels: Array<object>;
+  polling: boolean;
+  resourceId: string;
+  resourceLoaded: boolean;
+  resourceName: string;
+  resourceType: string;
+  rows: Array<TableRowType>;
+  selected: boolean;
+  selectedTable: string;
+  sidebarWidth: number;
+  tableNames: Array<string>;
+  totalRows: number;
+};
+
+function DataView({
+  canLink,
+  classes,
+  columns,
+  filterStatement,
+  handleChangeTable,
+  handleDrawerWidth,
+  pageBQQuery,
+  panels,
+  polling,
+  resourceId,
+  resourceLoaded,
+  resourceName,
+  resourceType,
+  rows,
+  selected,
+  selectedTable,
+  sidebarWidth,
+  tableNames,
+  totalRows,
+}: DataViewProps) {
+  const rowKey = (row: TableRowType): string => {
+    const drId = row.datarepo_row_id;
+    return drId ?? row.name;
+  };
+
+  // Can be removed after DR-2483
+  const showPanels = panels.length > 0;
+  // Only used for Direct BQ Query
+  const isDatasetFiltered = filterStatement.length > 0;
+
+  return (
+    //eslint-disable-next-line react/jsx-no-useless-fragment
+    <Fragment>
+      {resourceLoaded && (
+        <div className={classes.pageRoot}>
+          <AppBreadcrumbs
+            context={{
+              type:
+                resourceType === BreadcrumbType.DATASET
+                  ? BreadcrumbType.DATASET
+                  : BreadcrumbType.SNAPSHOT,
+              id: resourceId,
+              name: resourceName,
+            }}
+            childBreadcrumbs={[{ text: 'Data', to: 'data' }]}
+          />
+          <Typography variant="h3" className={classes.pageTitle}>
+            {resourceName}
+          </Typography>
+          <div className={classes.controls}>
+            <div className={classes.dropdown}>
+              <DataViewDropdown options={tableNames} onSelectedItem={handleChangeTable} />
+            </div>
+            <Link to={`/${resourceType}s/${resourceId}`}>
+              <Button
+                className={classes.viewDatasetButton}
+                color="primary"
+                variant="outlined"
+                disableElevation
+                size="large"
+              >
+                Back to Overview
+              </Button>
+            </Link>
+          </div>
+          <div className={showPanels ? classes.scrollTableWithPadding : classes.scrollTable}>
+            <LightTable
+              columns={columns}
+              filteredCount={totalRows}
+              loading={polling}
+              noRowsMessage={
+                isDatasetFiltered ? 'No rows match your filter' : 'No rows exist in the table'
+              }
+              pageBQQuery={pageBQQuery}
+              rowKey={rowKey}
+              rows={rows}
+              searchString={filterStatement}
+              totalCount={totalRows} // TODO - DR-2663 - instead should display total rows regardless of filtering
```

Comment:
Note: Ticket added to address TODO comment - https://broadworkbench.atlassian.net/browse/DR-2663